### PR TITLE
Add magiskboot to kernelsu working directory

### PR DESCRIPTION
**Pull request by fsxitutu**
Add the magiskboot binary to the /data/adb/ksu/bin directory so that scripts/programs can call magiskboot to patch the boot/init_boot image.

### CODE REVIEW

File: userspace/ksud/src/cli.rs [hunk 1]
```diff
@@ -307,7 +307,7 @@ pub fn run() -> Result<()> {
                 Module::Shrink => module::shrink_ksu_images(),
             }
         }
-        Commands::Install => utils::install(),
+        Commands::Install { magiskboot } => utils::install(magiskboot),
```
Comment (tiann) [inline userspace/ksud/src/cli.rs:310]: You should add cli parameter here: https://github.com/tiann/KernelSU/pull/1684/files#diff-50c7b7af6d0fd53a343c78c5b80654b053d158d7e3b24e744040165af53cc505R38

File: userspace/ksud/src/utils.rs [hunk 2]
```diff
@@ -208,10 +208,12 @@ fn link_ksud_to_bin() -> Result<()> {
     Ok(())
 }
 
-pub fn install() -> Result<()> {
+pub fn install(magiskboot: Option<PathBuf>) -> Result<()> {
     ensure_dir_exists(defs::ADB_DIR)?;
     std::fs::copy("/proc/self/exe", defs::DAEMON_PATH)?;
+    std::fs::copy(magiskboot, defs::MAGISKBOOT_PATH)?;
```
Comment (tiann) [inline userspace/ksud/src/utils.rs:214]: Please add the following two lines to the end of the function. This should not affect the existing logic if it fails.

**Review by tiann — CHANGES_REQUESTED** [review]

**Comment by tiann** [comment]
please fix the two issues, others look good to me

File: userspace/ksud/src/utils.rs [hunk 3]
```diff
@@ -219,6 +219,9 @@ pub fn install() -> Result<()> {
     link_ksud_to_bin()?;
 
     Ok(())
```
Comment (tiann) [inline userspace/ksud/src/utils.rs:221]: This should be the last one

**Review by tiann — CHANGES_REQUESTED** [review]

File: userspace/ksud/src/utils.rs (hunk 3 above)
Comment (fsxitutu) [reply userspace/ksud/src/utils.rs:232]: Capture possible errors to avoid affecting normal operation。utils.rs Already uploaded

**Review by tiann — APPROVED** [review]

**Pull request merged**